### PR TITLE
Gestis parser rewrite

### LINKED ISSUE #114: gestis get mak

### LINKED ISSUE #200: multiple cas numbers possible
- add field `comment` to CAS struct
-  substance example: cobaltnitrat (005340)

### LINKED ISSUE #993: Molar Mass not parsed from gestis
If I look up Calcium, the molar mass is not parsed.

### LINKED ISSUE #1052: Rewrite Gestis parser
- [x] write tests
- [x] rewrite and check
- [x] close all related issues
  - [x] #114 
  - [x] #200 
  - [x] #993

### PR DESCRIPTION
**Pull request by crapStone**
closes  #114
closes #200
closes #993
closes #1052

### CODE REVIEW

File: crates/search/contrib/gestis/helper.rs [hunk 1]
```diff
@@ -4,27 +4,63 @@ use std::{
...
+};
 
+/// This binary uses the `env_logger`.
+/// Just set the environment variable `RUST_LOG` to `debug` to see the debug output.
+///
+/// Other examples:
+///
+/// - RUST_LOG=search=trace -> show all log messages in search crate
+///
+/// - RUST_LOG=search::gestis::xml_parser=debug -> just show the debug messages of the xml parser
```
Comment (Epsilon02) [inline crates/search/contrib/gestis/helper.rs:22]: Maybe you should also mention this in one of our READMEs

File: crates/search/contrib/gestis/helper.rs [hunk 2]
```diff
@@ -4,27 +4,63 @@ use std::{
...
 
 #[tokio::main]
 async fn main() {
+  init_logger();
+
   let args = Arguments::from_args();
 
-  if args.extract.is_some() {
+  if args.parse.is_some() {
+    let gestis = gestis::Gestis::new(reqwest::ClientBuilder::new().build().unwrap());
```
Comment (Epsilon02) [inline crates/search/contrib/gestis/helper.rs:50]: Duplicated Code? (see line 58 in this file)

File: crates/search/src/gestis/mod.rs [hunk 3]
```diff
@@ -166,3 +167,21 @@ impl SearchType {
     }
   }
 }
+
+// TODO() remove again when SubstanceData was reworked
```
Comment (Epsilon02) [inline crates/search/src/gestis/mod.rs:171]: issue number

File: crates/search/src/gestis/xml_parser.rs [hunk 4]
```diff
@@ -1,652 +1,394 @@
...
+  This parser roughly does the following:
+  1. take the JSON response and extract references to all chapters we need by traversing all available chapters (`functions::parse_chapters`)
+  2. use a dedicated function for each chapter that extracts the necessary information with setup and error handling in `get_value`
+     (there may be a more complicated logic, e.g. mak parsing
+  3. pack all extracted information into a neat struct that can be used by the gestis handler to fill the SubstanceData struct
+
+  We don't trust gestis so every single XML/HTML string is searched until the end for everything that looks like stuff we could use.
+  This means, that every function starts with a definition of a `Vec` with some default capacity.
+  If there is not comment above or behind the value is the maximum we saw in the wild, so if you find a higher value
+  write change it (the values will be very low so the memory usage doesn't matter).
```
Comment (Epsilon02) [inline crates/search/src/gestis/xml_parser.rs:11]:
```suggestion
  If there is no comment above or behind the value it's the maximum we saw in the wild. So if you find a higher value,
  change it (the values will be very low so the memory usage doesn't matter).
```

File: crates/search/src/gestis/xml_parser.rs [hunk 5]
```diff
@@ -1,652 +1,394 @@
...
+
+  If you are the poor soul who has to fix a bug or change something because the gestis API has changed look in the
+  `contrib` folder in the search crate, I wrote a helper binary that has some useful features that eases the pain of
```
Comment (Epsilon02) [inline crates/search/src/gestis/xml_parser.rs:14]:
```suggestion
  `contrib` folder in the search crate. I wrote a helper binary that has some useful features that eases the pain of
```

File: crates/search/contrib/gestis/helper.rs (hunk 1 above)
Comment (crapStone) [reply crates/search/contrib/gestis/helper.rs:22]: This is just for the helper, we should add a complete explanation of the crates.
But not in this PR

File: crates/search/contrib/gestis/helper.rs (hunk 2 above)
Comment (crapStone) [reply crates/search/contrib/gestis/helper.rs:50]: yeah, but not all arguments need a Gestis object so imo this is ok

File: crates/search/src/gestis/mod.rs (hunk 3 above)
Comment (crapStone) [reply crates/search/src/gestis/mod.rs:171]: we still have to talk about what exactly we want to change

Comment (crapStone) [reply crates/search/src/gestis/mod.rs]: done

File: crates/logger/src/lib.rs [hunk 6]
```diff
@@ -26,9 +26,10 @@ pub async fn setup_logger() -> Result<(), fern::InitError> {
   Dispatch::new()
     .format(|out, message, record| {
       out.finish(format_args!(
-        "{}[{}][{}] {}",
-        chrono::Local::now().format("[%+]"),
+        "[{}][{}:{}][{}] {}",
```
Comment (6543) [inline crates/logger/src/lib.rs:29]: unrelated

File: frontend/src-wasm/Cargo.lock [hunk 7]
```diff
@@ -397,7 +397,7 @@ dependencies = [
  "chrono",
  "lazy_static",
  "log",
- "quick-xml",
+ "quick-xml 0.22.0",
```
Comment (6543) [inline frontend/src-wasm/Cargo.lock:400]: why is this version pinned?

Comment (crapStone) [reply frontend/src-wasm/Cargo.lock:400]: This is the lock file and because I used the newest alpha version there are now two versions of quick-xml in this lock file

File: crates/logger/src/lib.rs (hunk 6 above)
Comment (crapStone) [reply crates/logger/src/lib.rs:29]: Maybe, but this helped me debugging and I thought I can commit it here

**Review by Epsilon02 — APPROVED** [review]

**Review by 6543 — APPROVED** [review]
